edx_sga/tasks.py

```diff
@@ -0,0 +1,200 @@
+                loc=location,
+                sha1=answer['sha1'],
+                ext=os.path.splitext(answer['filename'])[1]
+            )
+
+            destination_file = os.path.join(
+                destination_path,
+                "{student_name}_{sha1}{ext}".format(
+                    student_name=student.username,
+                    sha1=answer['sha1'],
+                    ext=os.path.splitext(answer['filename'])[1]
+                )
+            )
+            try:
+                file_descriptor = default_storage.open(source_file)
+                with open(destination_file, 'wb') as file_pointer:
+                    while True:
+                        data = file_descriptor.read(BLOCK_SIZE)
+                        if data == '':  # end of file reached
+                            break
+                        file_pointer.write(data)
+
+            except IOError:
+                log.exception("Unable to download submission of student=%s", student.username)
+
+
+def _compress_folder(destination_path):
+    """
+    compress given folder/file
+
+    Args:
+        destination_path (str): path (including name) of folder/file which we want to compress.
+    """
+    make_archive(
+        destination_path,
+        'zip',
+        destination_path
+    )
+
+
+def _get_submissions_base_name(user_name, block_id, course_id):
+    """
+    returns submission folder name where we are saving files.
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        user_name (unicode): staff user name
+    """
+    return "{username}_submissions_{id}_{course_key}".format(
+        username=user_name,
+        id=hashlib.md5(block_id).hexdigest(),
+        course_key=course_id
+    )
+
+
+def _get_dir_path(submissions_dir_name, location=None):
+    """
+    returns directory path where we are saving files.
+
+    Args:
+        submissions_dir_name (str): name of folder
+        location (Location): location of sga module
+    """
+    if location:
+        return os.path.join(
+            DATA_DIR,
+            "{loc.org}/{loc.course}/{loc.run}/{submissions_dir_name}".format(
+                loc=location,
+                submissions_dir_name=submissions_dir_name
+            )
+        )
+
+    return os.path.join(DATA_DIR, submissions_dir_name)
+
+
+@CELERY_APP.task
+def zip_student_submissions(course_id, block_id, location, user):
+    """
+    Task to download all submissions as zip file
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user_name (str): staff user name
+    """
+    submissions_dir_name = _get_submissions_base_name(user.username, block_id, course_id)
+    destination_path = _get_dir_path(submissions_dir_name, location)
+    _mk_dir(destination_path)
+    remove_zip_file(user.username, block_id, course_id)
```

Comment:
I had an earlier comment about these two function calls ([link](https://github.com/mitodl/edx-sga/pull/187#discussion_r155655425)). Any thoughts on that?